src/keys/mod.rs

```diff
@@ -792,13 +792,18 @@ pub fn make_pkh<Pk: IntoDescriptorKey<Ctx>, Ctx: ScriptContext>(
 
 // Used internally by `bdk::fragment!` to build `multi()` fragments
 #[doc(hidden)]
-pub fn make_multi<Pk: IntoDescriptorKey<Ctx>, Ctx: ScriptContext>(
+pub fn make_multi<
+    Pk: IntoDescriptorKey<Ctx>,
+    Ctx: ScriptContext,
+    V: Fn(usize, Vec<DescriptorPublicKey>) -> Terminal<DescriptorPublicKey, Ctx>,
```

Comment:
Conceptual question.

The V here is a function but, in the usages of `make_multi` in dsl macros its used as, 
`$crate::keys::make_multi($thresh, $crate::miniscript::Terminal::Multi, $keys, &secp)`.

So `V` here is a `miniscript::Terminal::Multi` which is an enum..

How does this work??

Reply:
`Terminal::Multi` is not actually an enum, it's an enum *variant*. Just `Terminal` is the enum. Rust allows you to treat an enum variant as a function that creates an instance of that variant.

You may have used this pattern before when mapping errors since it's pretty common:

```rust
returns_a_result().map_err(MyErr::OtherVariant)
```

This is equivalent to

```rust
returns_a_result().map_err(|e| MyErr::OtherVariant(e))
```

Reply:
Oh this was interesting.. Cool stuff.. Thanks..